### PR TITLE
Add workflow for publishing model schemata to dandi/schema

### LINKED ISSUE #273: github workflow to mint .json rendering of the schema and populate dandi/schema
Should be published to https://github.com/dandi/schema/tree/master/releases/ - https://github.com/dandi/schema/blob/master/scripts/makeRelease.py is the original script used to produce those (and remove it from that repo)

- should be triggered only upon merge, ie. whenever PR which merges any models related stuff is merged and either modify already existing "released" file or create a new directory for a release based on the schema version

- Since we are in rapid development I do not think it is worth breeding new rc for minor changes, so it will be at our discretion to either boost it or not

### PR DESCRIPTION
**Pull request by jwodder**
Closes #273.

### CODE REVIEW

File: .github/workflows/publish-schemata.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+          token: ${{ secrets.DANDI_GITHUB_TOKEN }}
+
+      - name: Generate model schemata
+        run: python dandi/tools/pubschemata.py schema/releases
+
+      - name: Commit changes
+        run: |
+          git add releases
+          if ! git diff --quiet
+          then git commit -m "Publish model schemata as of dandi $(git -C ../dandi describe --tags)"
```
Comment (yarikoptic) [inline .github/workflows/publish-schemata.yml:45]:
```suggestion
          then git commit -m "Publish model schemata as of dandi-cli $(git -C ../dandi describe --tags)"
```

**Review by yarikoptic — CHANGES_REQUESTED** [review]
looks good and ok to merge after addressing few comments

File: .github/workflows/publish-schemata.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
+name: Publish model schemata
+
+on:
+  # ATM, this is the closest trigger to a PR merging
+  push:
+    branches:
+      - master
+    paths:
+      - dandi/models.py
+      - dandi/model_types.py
```
[the remark this answers is not in the document]
Comment (jwodder) [reply .github/workflows/publish-schemata.yml:10]: Done.

**Comment by yarikoptic** [comment]
Let's proceed!  Thank you @jwodder !

**Pull request merged**